### PR TITLE
feat(bundle-source): Endo symlinks and dev mode

### PR DESCRIPTION
**Pull request by kriskowal**
In this change, we sync the Endo repository published versions and thread two new features into bundleSource that are necessary to transition to using Zip files for archives generally.

1. The `moduleFormat` positional argument gets generalized to an options bag with `format` and `dev` fields. Endo uses the `dev` flag to allow `devDependencies` in an archive, which should only occur in tests if ever. The documentation dapp in particular needs either the `dev` flag or to use `devDependencies` more liberally.
2. Threads the additional `canonical` power that Endo needs to follow symbolic links. This is an ambient authority but can also now be overridden through injection into the existing `powers` argument @dckc set up. This keeps us in a position where we could factor out a powerless API layer from `bundleSource`, one that does not take powerful dependencies.

I elected to keep the testing light on this, since this is just threading a flag, but I welcome pressure to try harder there. The unit tests in the compartment mapper are here https://github.com/endojs/endo/blob/master/packages/compartment-mapper/test/test-main.js#L250-L294

### CODE REVIEW

**Comment by dckc** [comment]
> 1. The `moduleFormat` positional argument gets generalized to an options bag with `format` and `dev` fields.

Ouch... `moduleFormat` is used for duck-typing in various places; for example:

```
@param {{ moduleFormat: string, source: string }[]} bundles
```
https://github.com/Agoric/agoric-sdk/blob/8eba7f81fdd2e91f76ecb08ec2059961b2d05c5c/packages/SwingSet/src/controller.js#L50

```
@typedef { { moduleFormat: 'getExport', source: string } } ExportBundle
```
https://github.com/Agoric/agoric-sdk/blob/8eba7f81fdd2e91f76ecb08ec2059961b2d05c5c/packages/SwingSet/src/kernel/vatManager/manager-subprocess-xsnap.js#L31

```
@typedef {{ moduleFormat: unknown }} Bundle
```
https://github.com/Agoric/agoric-sdk/blob/8eba7f81fdd2e91f76ecb08ec2059961b2d05c5c/packages/SwingSet/src/types.js#L10

When I added that (or stuff like it) I didn't see any `types.js` in the bundle-source package, so I kind made stuff up as I went.

**Comment by kriskowal** [comment]
> > 1. The `moduleFormat` positional argument gets generalized to an options bag with `format` and `dev` fields.
> 
> Ouch... `moduleFormat` is used for duck-typing in various places; for example:
> 
> ```
> @param {{ moduleFormat: string, source: string }[]} bundles
> ```

From our out-of-band chat, this would be a problem with altering the shape of a bundle but is not so problematic for changing the shape of the bundleSource arguments.

File: packages/bundle-source/src/index.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ import * as babelParser from '@babel/parser';
 import babelGenerate from '@babel/generator';
 import babelTraverse from '@babel/traverse';
 import { makeArchive } from '@endo/compartment-mapper/archive.js';
+import { makeNodeReadPowers } from '@endo/compartment-mapper/src/node-powers.js';
```
Comment (michaelfig) [inline packages/bundle-source/src/index.js]: Be aware that we're trying to deprecate deep links into the `src` directory of a package.

**Review by michaelfig — APPROVED** [review]

File: packages/bundle-source/src/index.js (hunk 1 above)
Comment (kriskowal) [reply packages/bundle-source/src/index.js]: Thanks for catching this. I’d fixed the underlying problem with this morning’s publish and forgot to update this line.

**Pull request merged**